code/__DEFINES/traits.dm

```diff
@@ -114,6 +117,28 @@
 #define HAS_TRAIT_FROM_ONLY(target, trait, source) (target.status_traits?[trait] && (source in target.status_traits[trait]) && (length(target.status_traits[trait]) == 1))
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits?[trait] && (length(target.status_traits[trait] - source) > 0))
 
+// This trait gets added on a timer and then gets removed when the time is up
+// Adding the same trait twice while the timer is still active, resets the timer
+#define ADD_EXPIRING_TRAIT(target, trait, source, time) \
```

Comment:
This should be its own proc instead--it is unlikely to be as widely used as ADD_TRAIT.